<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
<title>org.junit.tests.assertion.AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch():void.d31fdf741990595d0121133ba36853da_node15</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<style type="text/css">
body { background-color:#fff; color:#000; font-family:Helvetica, sans-serif; font-size:10pt }
h1 { font-size:14pt }
.code { border-collapse:collapse; width:100%; font-family: "Andale Mono", monospace; font-size:10pt; line-height: 1.2em }
.color_black { color: black }
.color_blue { color: blue }
.color_green { color: green }
.color_orange { color: orange }
.color_red { color: red }
.line { border-left: 3px solid #ccc; padding-left: 1ex; white-space: pre; }
.msg { background-color:#fff8b4; color:#000; float:left; font-family:Helvetica, sans-serif; font-size: smaller; font-weight: bold; margin-bottom:10px; margin-top:10px; max-width:60em; padding:0.5em 1ex 0.5em 1ex; -webkit-border-radius:5px; -webkit-box-shadow:1px 1px 7px #000; white-space: pre-wrap; word-wrap: break-word; }
.num { background-color:#eee; color:#444; font-size: smaller; padding-right:2ex; text-align:right; width:2.5em; }
.dangling { background-color:white; color: gray; }
.visited { background-color:LemonChiffon; color: darkmagenta; }
.tooltip { display: none; background-color:#FFF0F0; border: 2px solid #F00; font-weight: normal; left:10em; padding: 2px; position: absolute; top: -1em; -webkit-border-radius:5px; -webkit-box-shadow:1px 1px 7px #000; z-index: 1}
.with_tooltip { position: relative; }
.with_tooltip:hover .tooltip, .visited:hover .tooltip { display: block; }
</style>
</head>
<body>
<center><h1>Cfg Node <a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE945">15</a></h1></center>PROC: <a href="./../org.junit.tests.assertion.AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch%28%29%3Avoid.d31fdf741990595d0121133ba36853da.html">void AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch()</a> LINE: <a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE945">945</a>
<br>PREDS:
<a href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch%28%29%3Avoid.d31fdf741990595d0121133ba36853da_node17.html"><span class='visited with_tooltip'>N_17<span class='tooltip'>node17 preds: succs:15 16 exn:3 </span></span></a><br>SUCCS:
<a href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch%28%29%3Avoid.d31fdf741990595d0121133ba36853da_node7.html"><span class='visited with_tooltip'>N_7<span class='tooltip'>node7 preds:15 succs:8 exn:3 </span></span></a><br>EXN:
<a href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch%28%29%3Avoid.d31fdf741990595d0121133ba36853da_node3.html"><span class='visited with_tooltip'>N_3<span class='tooltip'>node3 preds:16 succs:2 exn:2 </span></span></a><br>
<LISTING class='color_green'>n$23=_fun___instanceof(n$22:java.lang.AssertionError*,sizeof(t=java.lang.AssertionError):void) [line 945];
PRUNE(n$23, true); [line 945];
*&amp;CatchVar22:void=n$22 [line 945];
</LISTING>
<hr width="100%">
<a name="session20" href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch%28%29%3Avoid.d31fdf741990595d0121133ba36853da_node15.html#session20">node15#session20</a>(<a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE945">945</a>) [compute pre] liveness 15
<LISTING class='color_black'>JOIN

LEFT:   normal:{ CatchVar22, npe }, exn:{ }
RIGHT:  normal:{ }, exn:{ }
RESULT: normal:{ CatchVar22, npe }, exn:{ }

</LISTING>
<hr width="100%">
<a name="session21" href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch%28%29%3Avoid.d31fdf741990595d0121133ba36853da_node15.html#session21">node15#session21</a>(<a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE945">945</a>) [exec] liveness 15
<LISTING class='color_black'>PRE STATE:
normal:{ CatchVar22, npe }, exn:{ }

INSTR=  *&amp;CatchVar22:void=n$22 [line 945]; 

STATE:
normal:{ n$22, npe }, exn:{ }

INSTR=  PRUNE(n$23, true); [line 945]; 

STATE:
normal:{ n$22, n$23, npe }, exn:{ }

INSTR=  n$23=_fun___instanceof(n$22:java.lang.AssertionError*,sizeof(t=java.lang.AssertionError):void) [line 945]; 

STATE:
normal:{ n$22, npe }, exn:{ }

</LISTING>
<hr width="100%">
<a name="session47" href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch%28%29%3Avoid.d31fdf741990595d0121133ba36853da_node15.html#session47">node15#session47</a>(<a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE945">945</a>) [compute pre] nullify
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session48" href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch%28%29%3Avoid.d31fdf741990595d0121133ba36853da_node15.html#session48">node15#session48</a>(<a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE945">945</a>) [exec] nullify
<LISTING class='color_black'>PRE STATE:
(normal:{ n$22, npe }, exn:{ },normal:{ n$21, $irvar0, return }, exn:{ })

INSTR=  n$23=_fun___instanceof(n$22:java.lang.AssertionError*,sizeof(t=java.lang.AssertionError):void) [line 945]; 

STATE:
(normal:{ n$22, n$23, npe }, exn:{ },
normal:{ n$21, $irvar0, return }, exn:{ })

INSTR=  PRUNE(n$23, true); [line 945]; 

STATE UNCHANGED

INSTR=  *&amp;CatchVar22:void=n$22 [line 945]; 

STATE:
(normal:{ CatchVar22, npe }, exn:{ },normal:{ n$22, n$23 }, exn:{ })

</LISTING>
<hr width="100%">
<a name="session90" href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch%28%29%3Avoid.d31fdf741990595d0121133ba36853da_node15.html#session90">node15#session90</a>(<a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE945">945</a>) [exec] devirtualizer
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session91" href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch%28%29%3Avoid.d31fdf741990595d0121133ba36853da_node15.html#session91">node15#session91</a>(<a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE945">945</a>) [exec] devirtualizer
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session92" href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch%28%29%3Avoid.d31fdf741990595d0121133ba36853da_node15.html#session92">node15#session92</a>(<a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE945">945</a>) [exec] devirtualizer
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session93" href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch%28%29%3Avoid.d31fdf741990595d0121133ba36853da_node15.html#session93">node15#session93</a>(<a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE945">945</a>) [exec] devirtualizer
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session33" href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch%28%29%3Avoid.d31fdf741990595d0121133ba36853da_node15.html#session33">node15#session33</a>(<a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE945">945</a>) biabduction
<LISTING class='color_black'>**** [FP:Pending] void AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch() Node: 15, Procedure: void AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch(), Todo: 1 ****
  PROP 1 of 1:
  <span class='color_blue'><span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_red'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(n$22)</span>; <span class='color_red'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
  <span class='color_red'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_red'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>CatchVar22 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_red'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_red'>old_this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_red'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
  <span class='color_red'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_red'>n$22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
  [footprint
     <span class='color_red'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>  ]</span>
  .... Instructions: ....
  <span class='color_green'>n$23=_fun___instanceof(n$22:java.lang.AssertionError*,sizeof(t=java.lang.AssertionError):void) [line 945];
  PRUNE(n$23, true); [line 945];
  *&amp;CatchVar22:void=n$22 [line 945];
  EXIT_SCOPE(n$22,n$23); [line 945];
  </span>
  
  Processing prop 1/1
    Instruction <span class='color_green'>n$23=_fun___instanceof(n$22:java.lang.AssertionError*,sizeof(t=java.lang.AssertionError):void) [line 945]</span>
    in vpath_find exp:n$125
    vpath_find: cannot find n$125
    in vpath_find exp:n$119
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$22
    vpath_find: cannot find n$22
    <span class='color_orange'>WARNING: dropping non-footprint </span>n$22 != 0
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>n$23 = 1</span> ;
    <span class='color_blue'>n$22 != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(n$22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
    <span class='color_blue'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>n$22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    [footprint
       <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>  ]</span>
    
    Instruction <span class='color_green'>PRUNE(n$23, true); [line 945]</span>
    in vpath_find exp:n$133
    vpath_find: cannot find n$133
    in vpath_find exp:n$127
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$22
    vpath_find: cannot find n$22
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>n$23 = 1</span> ;
    <span class='color_blue'>n$22 != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(n$22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
    <span class='color_blue'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>n$22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    [footprint
       <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>  ]</span>
    
    Instruction <span class='color_green'>*&amp;CatchVar22:void=n$22 [line 945]</span>
    in vpath_find exp:n$141
    vpath_find: cannot find n$141
    in vpath_find exp:n$135
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$22
    vpath_find: cannot find n$22
    .... Rearrangement Start ....
    Exp: &amp;CatchVar22
    Prop:
    <span class='color_blue'>n$23 = 1 ;
    n$22 != 0; MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(n$141); MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe); MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(n$22); RET &lt; NullPointerException.<init>() &gt; (n$142) ;
    error = n$136<span class='color_orange'>initial</span>:java.lang.AssertionError*; $irvar3 = n$137<span class='color_orange'>initial</span>:void; CatchVar22 = n$138<span class='color_orange'>initial</span>:void; $irvar4 = n$139<span class='color_orange'>initial</span>:void; $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = null<span class='color_orange'>update:945</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$140<span class='color_orange'>initial</span>:void; this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); old_this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); npe = n$135<span class='color_orange'>update:941</span>:java.lang.NullPointerException* ;
    npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException; n$22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError
    [footprint
       this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )  ]</span>
    
    in prop_iter_extend_ptsto lexp: &amp;CatchVar22
    prop before:
    <span class='color_blue'>n$23 = 1 ;
    n$22 != 0; MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(n$141); MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe); MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(n$22); RET &lt; NullPointerException.<init>() &gt; (n$142) ;
    error = n$136<span class='color_orange'>initial</span>:java.lang.AssertionError*; $irvar3 = n$137<span class='color_orange'>initial</span>:void; CatchVar22 = n$138<span class='color_orange'>initial</span>:void; $irvar4 = n$139<span class='color_orange'>initial</span>:void; $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = null<span class='color_orange'>update:945</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$140<span class='color_orange'>initial</span>:void; this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); old_this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); npe = n$135<span class='color_orange'>update:941</span>:java.lang.NullPointerException* ;
    npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException; n$22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError
    [footprint
       this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )  ]</span>
    
    
    prop list after:
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>n$23 = 1</span> ;
    <span class='color_blue'>n$22 != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(n$141)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(n$22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (n$142)</span> ;
    <span class='color_blue'>error = n$136<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = n$137<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = n$138<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar4 = n$139<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = n$140<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = n$135<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>n$22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    [footprint
       <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>  ]</span>
    
    
    
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>n$23 = 1</span> ;
    <span class='color_blue'>CatchVar22 != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$4<span class='color_orange'>initial</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$3<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>n$22<span class='color_orange'>update:945</span></span>:void</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    [footprint
       <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>  ]</span>
    
    Instruction <span class='color_green'>EXIT_SCOPE(n$22,n$23); [line 945]</span>
    in vpath_find exp:_t$5
    vpath_find: cannot find _t$5
    in vpath_find exp:_t$0
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$22
    vpath_find do_sexp: no match on Eexp &amp;CatchVar22
    vpath_find: cannot find n$22
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>CatchVar22 != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$5<span class='color_orange'>initial</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$4<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>_t$3<span class='color_orange'>update:945</span></span>:void</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_red'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>
    [footprint
       <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>  ]</span>
    
    .... After Symbolic Execution ....
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>CatchVar22 != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
    <span class='color_blue'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = _t$3<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_red'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>
    [footprint
       <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>  ]</span>
    
    
    
  </LISTING>
<hr width="100%">
<a name="session49" href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch%28%29%3Avoid.d31fdf741990595d0121133ba36853da_node15.html#session49">node15#session49</a>(<a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE945">945</a>) biabduction
<LISTING class='color_black'>**** [RE:Pending] void AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch() Node: 15, Procedure: void AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch(), Todo: 1 ****
  PROP 1 of 1:
  <span class='color_blue'><span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_red'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(n$22)</span>; <span class='color_red'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
  <span class='color_red'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_red'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>CatchVar22 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$3<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_red'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_red'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_red'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_red'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_red'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
  <span class='color_red'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_red'>n$22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
  .... Instructions: ....
  <span class='color_green'>n$23=_fun___instanceof(n$22:java.lang.AssertionError*,sizeof(t=java.lang.AssertionError):void) [line 945];
  PRUNE(n$23, true); [line 945];
  *&amp;CatchVar22:void=n$22 [line 945];
  EXIT_SCOPE(n$22,n$23); [line 945];
  </span>
  
  Processing prop 1/1
    Instruction <span class='color_green'>n$23=_fun___instanceof(n$22:java.lang.AssertionError*,sizeof(t=java.lang.AssertionError):void) [line 945]</span>
    in vpath_find exp:n$125
    vpath_find: cannot find n$125
    in vpath_find exp:n$119
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$22
    vpath_find: cannot find n$22
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>n$23 = 1</span> ;
    <span class='color_blue'>n$22 != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(n$22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
    <span class='color_blue'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$3<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>n$22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>PRUNE(n$23, true); [line 945]</span>
    in vpath_find exp:n$133
    vpath_find: cannot find n$133
    in vpath_find exp:n$127
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$22
    vpath_find: cannot find n$22
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>n$23 = 1</span> ;
    <span class='color_blue'>n$22 != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(n$22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
    <span class='color_blue'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$3<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>n$22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>*&amp;CatchVar22:void=n$22 [line 945]</span>
    in vpath_find exp:n$141
    vpath_find: cannot find n$141
    in vpath_find exp:n$135
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$22
    vpath_find: cannot find n$22
    .... Rearrangement Start ....
    Exp: &amp;CatchVar22
    Prop:
    <span class='color_blue'>n$23 = 1 ;
    n$22 != 0; MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(n$141); MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe); MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(n$22); RET &lt; NullPointerException.<init>() &gt; (n$142) ;
    error = n$136<span class='color_orange'>initial</span>:java.lang.AssertionError*; $irvar3 = n$137<span class='color_orange'>initial</span>:void; CatchVar22 = n$138<span class='color_orange'>initial</span>:void; $RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$3<span class='color_orange'>formal</span>:java.lang.String*( sub ); $irvar4 = n$139<span class='color_orange'>initial</span>:void; $RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.Throwable*( sub ); $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = null<span class='color_orange'>update:945</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$140<span class='color_orange'>initial</span>:void; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); npe = n$135<span class='color_orange'>update:941</span>:java.lang.NullPointerException* ;
    npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException; n$22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>n$23 = 1</span> ;
    <span class='color_blue'>CatchVar22 != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$4<span class='color_orange'>initial</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$3<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>n$22<span class='color_orange'>update:945</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$3<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>EXIT_SCOPE(n$22,n$23); [line 945]</span>
    in vpath_find exp:_t$5
    vpath_find: cannot find _t$5
    in vpath_find exp:_t$0
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$22
    vpath_find do_sexp: no match on Eexp &amp;CatchVar22
    vpath_find: cannot find n$22
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>CatchVar22 != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$5<span class='color_orange'>initial</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$4<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>_t$3<span class='color_orange'>update:945</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$3<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_red'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span></span>
    
    .... After Symbolic Execution ....
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>CatchVar22 != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
    <span class='color_blue'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = _t$3<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$3<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_red'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span></span>
    
    
    
  </LISTING>
<hr width="100%">
<a name="session63" href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch%28%29%3Avoid.d31fdf741990595d0121133ba36853da_node15.html#session63">node15#session63</a>(<a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE945">945</a>) biabduction
<LISTING class='color_black'>**** [RE:Pending] void AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch() Node: 15, Procedure: void AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch(), Todo: 1 ****
  PROP 1 of 1:
  <span class='color_blue'><span class='color_red'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_red'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(n$22)</span>; <span class='color_red'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
  <span class='color_red'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_red'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>CatchVar22 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$3<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_red'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_red'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_red'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_red'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_red'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
  <span class='color_red'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_red'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar|->{}<span class='color_orange'>formal</span>:java.lang.String( sub )(instof)</span>; <span class='color_red'>n$22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
  .... Instructions: ....
  <span class='color_green'>n$23=_fun___instanceof(n$22:java.lang.AssertionError*,sizeof(t=java.lang.AssertionError):void) [line 945];
  PRUNE(n$23, true); [line 945];
  *&amp;CatchVar22:void=n$22 [line 945];
  EXIT_SCOPE(n$22,n$23); [line 945];
  </span>
  
  Processing prop 1/1
    Instruction <span class='color_green'>n$23=_fun___instanceof(n$22:java.lang.AssertionError*,sizeof(t=java.lang.AssertionError):void) [line 945]</span>
    in vpath_find exp:n$352
    vpath_find: cannot find n$352
    in vpath_find exp:n$346
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$22
    vpath_find: cannot find n$22
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>n$23 = 1</span> ;
    <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_blue'>n$22 != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(n$22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
    <span class='color_blue'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$3<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar|->{}<span class='color_orange'>formal</span>:java.lang.String( sub )(instof)</span>; <span class='color_blue'>n$22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>PRUNE(n$23, true); [line 945]</span>
    in vpath_find exp:n$360
    vpath_find: cannot find n$360
    in vpath_find exp:n$354
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$22
    vpath_find: cannot find n$22
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>n$23 = 1</span> ;
    <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_blue'>n$22 != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(n$22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
    <span class='color_blue'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$3<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar|->{}<span class='color_orange'>formal</span>:java.lang.String( sub )(instof)</span>; <span class='color_blue'>n$22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>*&amp;CatchVar22:void=n$22 [line 945]</span>
    in vpath_find exp:n$368
    vpath_find: cannot find n$368
    in vpath_find exp:n$362
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$22
    vpath_find: cannot find n$22
    .... Rearrangement Start ....
    Exp: &amp;CatchVar22
    Prop:
    <span class='color_blue'>n$23 = 1 ;
    $RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0; n$22 != 0; MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(n$368); MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe); MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(n$22); RET &lt; NullPointerException.<init>() &gt; (n$369) ;
    error = n$363<span class='color_orange'>initial</span>:java.lang.AssertionError*; $irvar3 = n$364<span class='color_orange'>initial</span>:void; CatchVar22 = n$365<span class='color_orange'>initial</span>:void; $RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$3<span class='color_orange'>formal</span>:java.lang.String*( sub ); $irvar4 = n$366<span class='color_orange'>initial</span>:void; $RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.Throwable*( sub ); $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = null<span class='color_orange'>update:945</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$367<span class='color_orange'>initial</span>:void; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); npe = n$362<span class='color_orange'>update:941</span>:java.lang.NullPointerException* ;
    $RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar|->{}<span class='color_orange'>formal</span>:java.lang.String( sub )(instof); npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException; n$22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>n$23 = 1</span> ;
    <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_blue'>CatchVar22 != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$4<span class='color_orange'>initial</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$3<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>n$22<span class='color_orange'>update:945</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$3<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar|->{}<span class='color_orange'>formal</span>:java.lang.String( sub )(instof)</span>; <span class='color_blue'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>EXIT_SCOPE(n$22,n$23); [line 945]</span>
    in vpath_find exp:_t$5
    vpath_find: cannot find _t$5
    in vpath_find exp:_t$0
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$22
    vpath_find do_sexp: no match on Eexp &amp;CatchVar22
    vpath_find: cannot find n$22
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>CatchVar22 != 0</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$5<span class='color_orange'>initial</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$4<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>_t$3<span class='color_orange'>update:945</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$3<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_red'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar|->{}<span class='color_orange'>formal</span>:java.lang.String( sub )(instof)</span></span>
    
    .... After Symbolic Execution ....
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>CatchVar22 != 0</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
    <span class='color_blue'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = _t$3<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$3<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_red'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar|->{}<span class='color_orange'>formal</span>:java.lang.String( sub )(instof)</span></span>
    
    
    
  </LISTING>
<hr width="100%">
<a name="session77" href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch%28%29%3Avoid.d31fdf741990595d0121133ba36853da_node15.html#session77">node15#session77</a>(<a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE945">945</a>) biabduction
<LISTING class='color_black'>**** [RE:Pending] void AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch() Node: 15, Procedure: void AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch(), Todo: 1 ****
  PROP 1 of 1:
  <span class='color_blue'><span class='color_red'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_red'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(n$22)</span>; <span class='color_red'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
  <span class='color_red'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_red'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>CatchVar22 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_red'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_red'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_red'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_red'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
  <span class='color_red'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_red'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar|->{}<span class='color_orange'>formal</span>:java.lang.String( sub )(instof)</span>; <span class='color_red'>n$22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
  .... Instructions: ....
  <span class='color_green'>n$23=_fun___instanceof(n$22:java.lang.AssertionError*,sizeof(t=java.lang.AssertionError):void) [line 945];
  PRUNE(n$23, true); [line 945];
  *&amp;CatchVar22:void=n$22 [line 945];
  EXIT_SCOPE(n$22,n$23); [line 945];
  </span>
  
  Processing prop 1/1
    Instruction <span class='color_green'>n$23=_fun___instanceof(n$22:java.lang.AssertionError*,sizeof(t=java.lang.AssertionError):void) [line 945]</span>
    in vpath_find exp:n$579
    vpath_find: cannot find n$579
    in vpath_find exp:n$573
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$22
    vpath_find: cannot find n$22
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>n$23 = 1</span> ;
    <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_blue'>n$22 != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(n$22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
    <span class='color_blue'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar|->{}<span class='color_orange'>formal</span>:java.lang.String( sub )(instof)</span>; <span class='color_blue'>n$22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>PRUNE(n$23, true); [line 945]</span>
    in vpath_find exp:n$587
    vpath_find: cannot find n$587
    in vpath_find exp:n$581
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$22
    vpath_find: cannot find n$22
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>n$23 = 1</span> ;
    <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_blue'>n$22 != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(n$22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
    <span class='color_blue'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar|->{}<span class='color_orange'>formal</span>:java.lang.String( sub )(instof)</span>; <span class='color_blue'>n$22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>*&amp;CatchVar22:void=n$22 [line 945]</span>
    in vpath_find exp:n$595
    vpath_find: cannot find n$595
    in vpath_find exp:n$589
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$22
    vpath_find: cannot find n$22
    .... Rearrangement Start ....
    Exp: &amp;CatchVar22
    Prop:
    <span class='color_blue'>n$23 = 1 ;
    $RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0; n$22 != 0; MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(n$595); MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe); MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(n$22); RET &lt; NullPointerException.<init>() &gt; (n$596) ;
    error = n$590<span class='color_orange'>initial</span>:java.lang.AssertionError*; $irvar3 = n$591<span class='color_orange'>initial</span>:void; CatchVar22 = n$592<span class='color_orange'>initial</span>:void; $RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.String*( sub ); $irvar4 = n$593<span class='color_orange'>initial</span>:void; $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = null<span class='color_orange'>update:945</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$594<span class='color_orange'>initial</span>:void; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); npe = n$589<span class='color_orange'>update:941</span>:java.lang.NullPointerException* ;
    $RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar|->{}<span class='color_orange'>formal</span>:java.lang.String( sub )(instof); npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException; n$22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>n$23 = 1</span> ;
    <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_blue'>CatchVar22 != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$4<span class='color_orange'>initial</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$3<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>n$22<span class='color_orange'>update:945</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar|->{}<span class='color_orange'>formal</span>:java.lang.String( sub )(instof)</span>; <span class='color_blue'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>EXIT_SCOPE(n$22,n$23); [line 945]</span>
    in vpath_find exp:_t$5
    vpath_find: cannot find _t$5
    in vpath_find exp:_t$0
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$22
    vpath_find do_sexp: no match on Eexp &amp;CatchVar22
    vpath_find: cannot find n$22
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>CatchVar22 != 0</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$5<span class='color_orange'>initial</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$4<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>_t$3<span class='color_orange'>update:945</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_red'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar|->{}<span class='color_orange'>formal</span>:java.lang.String( sub )(instof)</span></span>
    
    .... After Symbolic Execution ....
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>CatchVar22 != 0</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
    <span class='color_blue'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = _t$3<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_red'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar|->{}<span class='color_orange'>formal</span>:java.lang.String( sub )(instof)</span></span>
    
    
    
  </LISTING>
<hr width="100%">
<a name="session91" href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch%28%29%3Avoid.d31fdf741990595d0121133ba36853da_node15.html#session91">node15#session91</a>(<a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE945">945</a>) biabduction
<LISTING class='color_black'>**** [RE:Pending] void AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch() Node: 15, Procedure: void AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch(), Todo: 1 ****
  PROP 1 of 1:
  <span class='color_blue'><span class='color_red'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_red'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(n$22)</span>; <span class='color_red'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
  <span class='color_red'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_red'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>CatchVar22 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_red'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_red'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_red'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_red'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
  <span class='color_red'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_red'>n$22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
  .... Instructions: ....
  <span class='color_green'>n$23=_fun___instanceof(n$22:java.lang.AssertionError*,sizeof(t=java.lang.AssertionError):void) [line 945];
  PRUNE(n$23, true); [line 945];
  *&amp;CatchVar22:void=n$22 [line 945];
  EXIT_SCOPE(n$22,n$23); [line 945];
  </span>
  
  Processing prop 1/1
    Instruction <span class='color_green'>n$23=_fun___instanceof(n$22:java.lang.AssertionError*,sizeof(t=java.lang.AssertionError):void) [line 945]</span>
    in vpath_find exp:n$808
    vpath_find: cannot find n$808
    in vpath_find exp:n$802
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$22
    vpath_find: cannot find n$22
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>n$23 = 1</span> ;
    <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_blue'>n$22 != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(n$22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
    <span class='color_blue'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>n$22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>PRUNE(n$23, true); [line 945]</span>
    in vpath_find exp:n$816
    vpath_find: cannot find n$816
    in vpath_find exp:n$810
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$22
    vpath_find: cannot find n$22
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>n$23 = 1</span> ;
    <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_blue'>n$22 != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(n$22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
    <span class='color_blue'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>n$22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>*&amp;CatchVar22:void=n$22 [line 945]</span>
    in vpath_find exp:n$824
    vpath_find: cannot find n$824
    in vpath_find exp:n$818
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$22
    vpath_find: cannot find n$22
    .... Rearrangement Start ....
    Exp: &amp;CatchVar22
    Prop:
    <span class='color_blue'>n$23 = 1 ;
    $RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0; n$22 != 0; MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(n$824); MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe); MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(n$22); RET &lt; NullPointerException.<init>() &gt; (n$825) ;
    error = n$819<span class='color_orange'>initial</span>:java.lang.AssertionError*; $irvar3 = n$820<span class='color_orange'>initial</span>:void; CatchVar22 = n$821<span class='color_orange'>initial</span>:void; $RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.String*( sub ); $irvar4 = n$822<span class='color_orange'>initial</span>:void; $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = null<span class='color_orange'>update:945</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$823<span class='color_orange'>initial</span>:void; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); npe = n$818<span class='color_orange'>update:941</span>:java.lang.NullPointerException* ;
    npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException; n$22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>n$23 = 1</span> ;
    <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_blue'>CatchVar22 != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$4<span class='color_orange'>initial</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$3<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>n$22<span class='color_orange'>update:945</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>EXIT_SCOPE(n$22,n$23); [line 945]</span>
    in vpath_find exp:_t$5
    vpath_find: cannot find _t$5
    in vpath_find exp:_t$0
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$22
    vpath_find do_sexp: no match on Eexp &amp;CatchVar22
    vpath_find: cannot find n$22
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>CatchVar22 != 0</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$5<span class='color_orange'>initial</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$4<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>_t$3<span class='color_orange'>update:945</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_red'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span></span>
    
    .... After Symbolic Execution ....
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>CatchVar22 != 0</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
    <span class='color_blue'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = _t$3<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_red'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span></span>
    
    
    
  </LISTING>
<hr width="100%">
<a name="session105" href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch%28%29%3Avoid.d31fdf741990595d0121133ba36853da_node15.html#session105">node15#session105</a>(<a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE945">945</a>) biabduction
<LISTING class='color_black'>**** [RE:Pending] void AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch() Node: 15, Procedure: void AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch(), Todo: 1 ****
  PROP 1 of 1:
  <span class='color_blue'><span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_red'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(n$22)</span>; <span class='color_red'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
  <span class='color_red'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_red'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>CatchVar22 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = 0<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_red'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_red'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_red'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_red'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
  <span class='color_red'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_red'>n$22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
  .... Instructions: ....
  <span class='color_green'>n$23=_fun___instanceof(n$22:java.lang.AssertionError*,sizeof(t=java.lang.AssertionError):void) [line 945];
  PRUNE(n$23, true); [line 945];
  *&amp;CatchVar22:void=n$22 [line 945];
  EXIT_SCOPE(n$22,n$23); [line 945];
  </span>
  
  Processing prop 1/1
    Instruction <span class='color_green'>n$23=_fun___instanceof(n$22:java.lang.AssertionError*,sizeof(t=java.lang.AssertionError):void) [line 945]</span>
    in vpath_find exp:n$1037
    vpath_find: cannot find n$1037
    in vpath_find exp:n$1031
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$22
    vpath_find: cannot find n$22
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>n$23 = 1</span> ;
    <span class='color_blue'>n$22 != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(n$22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
    <span class='color_blue'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = 0<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>n$22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>PRUNE(n$23, true); [line 945]</span>
    in vpath_find exp:n$1045
    vpath_find: cannot find n$1045
    in vpath_find exp:n$1039
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$22
    vpath_find: cannot find n$22
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>n$23 = 1</span> ;
    <span class='color_blue'>n$22 != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(n$22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
    <span class='color_blue'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = 0<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>n$22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>*&amp;CatchVar22:void=n$22 [line 945]</span>
    in vpath_find exp:n$1053
    vpath_find: cannot find n$1053
    in vpath_find exp:n$1047
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$22
    vpath_find: cannot find n$22
    .... Rearrangement Start ....
    Exp: &amp;CatchVar22
    Prop:
    <span class='color_blue'>n$23 = 1 ;
    n$22 != 0; MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(n$1053); MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe); MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(n$22); RET &lt; NullPointerException.<init>() &gt; (n$1054) ;
    error = n$1048<span class='color_orange'>initial</span>:java.lang.AssertionError*; $irvar3 = n$1049<span class='color_orange'>initial</span>:void; CatchVar22 = n$1050<span class='color_orange'>initial</span>:void; $RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = 0<span class='color_orange'>formal</span>:java.lang.String*( sub ); $irvar4 = n$1051<span class='color_orange'>initial</span>:void; $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = null<span class='color_orange'>update:945</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$1052<span class='color_orange'>initial</span>:void; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); npe = n$1047<span class='color_orange'>update:941</span>:java.lang.NullPointerException* ;
    npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException; n$22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>n$23 = 1</span> ;
    <span class='color_blue'>CatchVar22 != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$4<span class='color_orange'>initial</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$3<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>n$22<span class='color_orange'>update:945</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = 0<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>EXIT_SCOPE(n$22,n$23); [line 945]</span>
    in vpath_find exp:_t$5
    vpath_find: cannot find _t$5
    in vpath_find exp:_t$0
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$22
    vpath_find do_sexp: no match on Eexp &amp;CatchVar22
    vpath_find: cannot find n$22
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>CatchVar22 != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$5<span class='color_orange'>initial</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$4<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>_t$3<span class='color_orange'>update:945</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = 0<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_red'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span></span>
    
    .... After Symbolic Execution ....
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>CatchVar22 != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
    <span class='color_blue'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = _t$3<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = 0<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_red'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span></span>
    
    
    
  </LISTING>
<hr width="100%">
<a name="session119" href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch%28%29%3Avoid.d31fdf741990595d0121133ba36853da_node15.html#session119">node15#session119</a>(<a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE945">945</a>) biabduction
<LISTING class='color_black'>**** [RE:Pending] void AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch() Node: 15, Procedure: void AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch(), Todo: 1 ****
  PROP 1 of 1:
  <span class='color_blue'><span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_red'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(n$22)</span>; <span class='color_red'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
  <span class='color_red'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_red'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>CatchVar22 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = 0<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_red'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$5<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_red'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_red'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>this = val$4<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_red'>old_this = val$4<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_red'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
  <span class='color_red'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_red'>n$22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
  .... Instructions: ....
  <span class='color_green'>n$23=_fun___instanceof(n$22:java.lang.AssertionError*,sizeof(t=java.lang.AssertionError):void) [line 945];
  PRUNE(n$23, true); [line 945];
  *&amp;CatchVar22:void=n$22 [line 945];
  EXIT_SCOPE(n$22,n$23); [line 945];
  </span>
  
  Processing prop 1/1
    Instruction <span class='color_green'>n$23=_fun___instanceof(n$22:java.lang.AssertionError*,sizeof(t=java.lang.AssertionError):void) [line 945]</span>
    in vpath_find exp:n$1266
    vpath_find: cannot find n$1266
    in vpath_find exp:n$1260
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$22
    vpath_find: cannot find n$22
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>n$23 = 1</span> ;
    <span class='color_blue'>n$22 != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(n$22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
    <span class='color_blue'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = 0<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$5<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$4<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$4<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>n$22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>PRUNE(n$23, true); [line 945]</span>
    in vpath_find exp:n$1274
    vpath_find: cannot find n$1274
    in vpath_find exp:n$1268
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$22
    vpath_find: cannot find n$22
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>n$23 = 1</span> ;
    <span class='color_blue'>n$22 != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(n$22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
    <span class='color_blue'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = 0<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$5<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$4<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$4<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>n$22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>*&amp;CatchVar22:void=n$22 [line 945]</span>
    in vpath_find exp:n$1282
    vpath_find: cannot find n$1282
    in vpath_find exp:n$1276
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$22
    vpath_find: cannot find n$22
    .... Rearrangement Start ....
    Exp: &amp;CatchVar22
    Prop:
    <span class='color_blue'>n$23 = 1 ;
    n$22 != 0; MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(n$1282); MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe); MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(n$22); RET &lt; NullPointerException.<init>() &gt; (n$1283) ;
    error = n$1277<span class='color_orange'>initial</span>:java.lang.AssertionError*; $irvar3 = n$1278<span class='color_orange'>initial</span>:void; CatchVar22 = n$1279<span class='color_orange'>initial</span>:void; $RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = 0<span class='color_orange'>formal</span>:java.lang.String*( sub ); $irvar4 = n$1280<span class='color_orange'>initial</span>:void; $RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$5<span class='color_orange'>formal</span>:java.lang.Throwable*( sub ); $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = null<span class='color_orange'>update:945</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$1281<span class='color_orange'>initial</span>:void; this = val$4<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); old_this = val$4<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); npe = n$1276<span class='color_orange'>update:941</span>:java.lang.NullPointerException* ;
    npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException; n$22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>n$23 = 1</span> ;
    <span class='color_blue'>CatchVar22 != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$4<span class='color_orange'>initial</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$3<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>n$22<span class='color_orange'>update:945</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = 0<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$5<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$4<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$4<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>EXIT_SCOPE(n$22,n$23); [line 945]</span>
    in vpath_find exp:_t$5
    vpath_find: cannot find _t$5
    in vpath_find exp:_t$0
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$22
    vpath_find do_sexp: no match on Eexp &amp;CatchVar22
    vpath_find: cannot find n$22
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>CatchVar22 != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$5<span class='color_orange'>initial</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$4<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>_t$3<span class='color_orange'>update:945</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = 0<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$5<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$4<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$4<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_red'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span></span>
    
    .... After Symbolic Execution ....
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>CatchVar22 != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
    <span class='color_blue'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = _t$3<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = 0<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$5<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$4<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$4<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_red'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span></span>
    
    
    
  </LISTING>
<hr width="100%">
<a name="session133" href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch%28%29%3Avoid.d31fdf741990595d0121133ba36853da_node15.html#session133">node15#session133</a>(<a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE945">945</a>) biabduction
<LISTING class='color_black'>**** [RE:Pending] void AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch() Node: 15, Procedure: void AssertionTest.assertThrowsUsesCanonicalNameUponTypeMismatch(), Todo: 1 ****
  PROP 1 of 1:
  <span class='color_blue'><span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_red'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(n$22)</span>; <span class='color_red'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
  <span class='color_red'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_red'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>CatchVar22 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$8<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_red'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$7<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_red'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_red'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>this = val$6<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_red'>old_this = val$6<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_red'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
  <span class='color_red'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_red'>n$22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
  .... Instructions: ....
  <span class='color_green'>n$23=_fun___instanceof(n$22:java.lang.AssertionError*,sizeof(t=java.lang.AssertionError):void) [line 945];
  PRUNE(n$23, true); [line 945];
  *&amp;CatchVar22:void=n$22 [line 945];
  EXIT_SCOPE(n$22,n$23); [line 945];
  </span>
  
  Processing prop 1/1
    Instruction <span class='color_green'>n$23=_fun___instanceof(n$22:java.lang.AssertionError*,sizeof(t=java.lang.AssertionError):void) [line 945]</span>
    in vpath_find exp:n$1493
    vpath_find: cannot find n$1493
    in vpath_find exp:n$1487
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$22
    vpath_find: cannot find n$22
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>n$23 = 1</span> ;
    <span class='color_blue'>n$22 != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(n$22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
    <span class='color_blue'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$8<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$7<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$6<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$6<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>n$22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>PRUNE(n$23, true); [line 945]</span>
    in vpath_find exp:n$1501
    vpath_find: cannot find n$1501
    in vpath_find exp:n$1495
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$22
    vpath_find: cannot find n$22
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>n$23 = 1</span> ;
    <span class='color_blue'>n$22 != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(n$22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
    <span class='color_blue'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$8<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$7<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$6<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$6<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>n$22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>*&amp;CatchVar22:void=n$22 [line 945]</span>
    in vpath_find exp:n$1509
    vpath_find: cannot find n$1509
    in vpath_find exp:n$1503
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$22
    vpath_find: cannot find n$22
    .... Rearrangement Start ....
    Exp: &amp;CatchVar22
    Prop:
    <span class='color_blue'>n$23 = 1 ;
    n$22 != 0; MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(n$1509); MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe); MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(n$22); RET &lt; NullPointerException.<init>() &gt; (n$1510) ;
    error = n$1504<span class='color_orange'>initial</span>:java.lang.AssertionError*; $irvar3 = n$1505<span class='color_orange'>initial</span>:void; CatchVar22 = n$1506<span class='color_orange'>initial</span>:void; $RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$8<span class='color_orange'>formal</span>:java.lang.String*( sub ); $irvar4 = n$1507<span class='color_orange'>initial</span>:void; $RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$7<span class='color_orange'>formal</span>:java.lang.Throwable*( sub ); $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = null<span class='color_orange'>update:945</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$1508<span class='color_orange'>initial</span>:void; this = val$6<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); old_this = val$6<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); npe = n$1503<span class='color_orange'>update:941</span>:java.lang.NullPointerException* ;
    npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException; n$22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>n$23 = 1</span> ;
    <span class='color_blue'>CatchVar22 != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$4<span class='color_orange'>initial</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$3<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>n$22<span class='color_orange'>update:945</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$8<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$7<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$6<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$6<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>EXIT_SCOPE(n$22,n$23); [line 945]</span>
    in vpath_find exp:_t$5
    vpath_find: cannot find _t$5
    in vpath_find exp:_t$0
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$22
    vpath_find do_sexp: no match on Eexp &amp;CatchVar22
    vpath_find: cannot find n$22
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>CatchVar22 != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$5<span class='color_orange'>initial</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$4<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>_t$3<span class='color_orange'>update:945</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$8<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$7<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$6<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$6<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_red'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span></span>
    
    .... After Symbolic Execution ....
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>CatchVar22 != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):944 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):944 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>MEMne &lt; new:941 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
    <span class='color_blue'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = _t$3<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$8<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$7<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:945</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$6<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$6<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:941</span>:java.lang.NullPointerException*</span> ;
    <span class='color_red'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span></span>
    
    
    
  </LISTING>